configure

```diff
@@ -21,8 +21,44 @@ else
 fi
 
 
+# By defalt CXX_STD is CXX11
+CXX_STD=CXX11
+
+# But if it works, use CXX17
+CXX=`"${R_HOME}"/bin/R CMD config CXX`
+CXX17STD=`"${R_HOME}"/bin/R CMD config CXX17STD`
+CXX17FLAGS=`"${R_HOME}"/bin/R CMD config CXX17FLAGS`
+if [ "$CXX" != "" -a "$CXX17FLAGS" != "" -a "$CXX17STD" != "" ]; then
+        ptimespec='#include <cstdio>
+#include <ctime>
+// Example adapted from https://en.cppreference.com/w/cpp/chrono/c/timespec_get
+
+namespace std {
+
+        int main()
+        {
+                timespec ts;
+                timespec_get(&ts, TIME_UTC);
+                char buf[100];
+                strftime(buf, sizeof buf, "%D %T", gmtime(&ts.tv_sec));
+                printf("Current time: %s.%09ld UTC", buf, ts.tv_nsec);
+                return 0;
+        }
+
+}'
+        echo "$ptimespec" | ${CXX} -x c++ -c $CXX17FLAGS $CXX17STD - -o /dev/null > /dev/null 2>&1
+        if [ $? -eq 0 ]; then
+                # See https://github.com/r-lib/later/issues/115
+                echo "Using c++17."
+                CXX_STD=CXX17
+        else
+                echo "Cannot use c++17 using default configuration for c++11"
```

Comment:
It's a weird thing about R, that when one sets `CXX_STD=CXX11`, it doesn't necessarily mean that it will be compiled with a C++11 compiler. It just means that if a C++11 compiler is available, it will use it. (Note that this is different from how R uses `CXX14` and `CXX17`.)